copy/copy.go

```diff
@@ -79,8 +80,10 @@ func Copy(ctx context.Context, srcRoot, src, dstRoot, dst string, opts ...Opt) e
 		if err != nil {
 			return err
 		}
-		if err := MkdirAll(ensureDstPath, 0755, ci.Chown, ci.Utime); err != nil {
+		if createdDirs, err := MkdirAll(ensureDstPath, 0755, ci.Chown, ci.Utime); err != nil {
 			return err
+		} else {
+			defer fixCreatedParentDirs(createdDirs, ci.Utime)
```

Comment:
Related to https://github.com/docker/buildx/pull/2745#discussion_r1810862520, could we apply ownership as well?:

```suggestion
			defer fixCreatedParentDirs(createdDirs, ci.Chown, ci.Utime)
```

Reply:
The issue is that although target directories are created with requested timestamp, when files are written into them it changes the modified time. So the modified time needs to be reset again after the copy. This problem does not exist for chown.

There is another weird case in this code where target dirs are always created with default `755` mode, even if `--chmod` is set. Not sure if this is correct, but not directly part of this issue.